### ISSUE TITLE
The content generation command for "SRC:-build/qemu-gui_autogen/moc_predefs.h" failed.

### ISSUE THREAD

**Issue by username7778** [ispras/swat#7]
hello

i get this error below. do i miss a package?
maybe someone can help?

andre@DESKTOP-6FLKL1B MINGW64 ~
$ cd qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ cmake ../qemu-gui
System is unknown to cmake, create:
Platform/MINGW64_NT-10.0-19043 to use this system, please post your config file on discourse.cmake.org so it can be added to cmake
Your CMakeCache.txt file was copied to CopyOfCMakeCache.txt. Please post that file on discourse.cmake.org.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/andre/qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ make ./qemu-gui
[  2%] Automatic MOC for target qemu-gui

AutoMoc subprocess error
------------------------
The content generation command for "SRC:-build/qemu-gui_autogen/moc_predefs.h" failed.

Command
-------
/mingw64/bin/CC.exe -dM -E -c /usr/share/cmake-3.23.2/Modules/CMakeCXXCompilerABI.cpp -DGUI -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_WIDGETS_LIB -I/home/andre/qemu-gui-build -I/home/andre/qemu-gui -I/home/andre/qemu-gui/gui -I/home/andre/qemu-gui/device -I/home/andre/qemu-gui/config -I/home/andre/qemu-gui/qemu -I/mingw64/include -I/mingw64/include/QtWidgets -I/mingw64/include/QtGui -I/mingw64/include/QtCore -I/mingw64/share/qt5/mkspecs/win32-g++ -I/mingw64/include/QtNetwork

Output
------

make[3]: *** [CMakeFiles/qemu-gui_autogen.dir/build.make:71: CMakeFiles/qemu-gui_autogen] Error 1
make[2]: *** [CMakeFiles/Makefile2:139: CMakeFiles/qemu-gui_autogen.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:94: CMakeFiles/qemu-gui.dir/rule] Error 2
make: *** [Makefile:169: qemu-gui] Error 2

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build

**Comment by Dovgalyuk** [comment]
According to this, there is some problem with the compiler:

System is unknown to cmake, create:
Platform/MINGW64_NT-10.0-19043 to use this system, please post your config file on discourse.cmake.org so it can be added to cmake

Do you have any C++ compiler installed in MinGW?

**Comment by username7778** [comment]
yes i have mingw-w64-x86_64-gcc-libs-12.2.0-4 as c/c++ compiler.

i removed cmake and make (pacman -R cmake pacman -R make)
and insted used pacman -S mingw-w64-x86_64-cmake pacman -S mingw-w64-x86_64-make
i got following output

andre@DESKTOP-6FLKL1B MINGW64 ~
$ mkdir qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~
$ cd qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ cmake ../qemu-gui
-- Building for: Ninja
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Users/andre/Desktop/mysys/mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Users/andre/Desktop/mysys/mingw64/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/andre/Desktop/mysys/home/andre/qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ make ./qemu-gui
make: *** No rule to make target 'qemu-gui'.  Stop.

**Comment by Dovgalyuk** [comment]
try just "make" instead of last command

**Comment by username7778** [comment]
hello

output with mingw-w64-x86_64-cmake and mingw-w64-x86_64-make

andre@DESKTOP-6FLKL1B MINGW64 ~
$ cd qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ cmake ../qemu-gui
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/andre/Desktop/mysys/home/andre/qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ make
make: *** No targets specified and no makefile found.  Stop.

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build

output when mingw-w64-x86_64-cmake and mingw-w64-x86_64-make are replaced with  cmake and make

andre@DESKTOP-6FLKL1B MINGW64 ~
$ git clone https://github.com/ispras/qemu-gui
Cloning into 'qemu-gui'...
remote: Enumerating objects: 2207, done.
remote: Counting objects: 100% (363/363), done.
remote: Compressing objects: 100% (120/120), done.
remote: Total 2207 (delta 241), reused 342 (delta 231), pack-reused 1844
Receiving objects: 100% (2207/2207), 653.73 KiB | 7.43 MiB/s, done.
Resolving deltas: 100% (1617/1617), done.

andre@DESKTOP-6FLKL1B MINGW64 ~
$ mkdir qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~
$ cd qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ cmake ../qemu-gui
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
System is unknown to cmake, create:
Platform/MINGW64_NT-10.0-19043 to use this system, please post your config file on discourse.cmake.org so it can be added to cmake
-- Detecting C compiler ABI info
System is unknown to cmake, create:
Platform/MINGW64_NT-10.0-19043 to use this system, please post your config file on discourse.cmake.org so it can be added to cmake
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
System is unknown to cmake, create:
Platform/MINGW64_NT-10.0-19043 to use this system, please post your config file on discourse.cmake.org so it can be added to cmake
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /mingw64/bin/CC.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/andre/qemu-gui-build

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build
$ make
[  1%] Automatic MOC for target qemu-gui

AutoMoc subprocess error
------------------------
The content generation command for "SRC:-build/qemu-gui_autogen/moc_predefs.h" failed.

Command
-------
/mingw64/bin/CC.exe -dM -E -c /usr/share/cmake-3.23.2/Modules/CMakeCXXCompilerABI.cpp -DGUI -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_WIDGETS_LIB -I/home/andre/qemu-gui-build -I/home/andre/qemu-gui -I/home/andre/qemu-gui/gui -I/home/andre/qemu-gui/device -I/home/andre/qemu-gui/config -I/home/andre/qemu-gui/qemu -I/mingw64/include -I/mingw64/include/QtWidgets -I/mingw64/include/QtGui -I/mingw64/include/QtCore -I/mingw64/share/qt5/mkspecs/win32-g++ -I/mingw64/include/QtNetwork

Output
------


make[2]: *** [CMakeFiles/qemu-gui_autogen.dir/build.make:71: CMakeFiles/qemu-gui_autogen] Error 1
make[1]: *** [CMakeFiles/Makefile2:139: CMakeFiles/qemu-gui_autogen.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

andre@DESKTOP-6FLKL1B MINGW64 ~/qemu-gui-build

**Comment by Dovgalyuk** [comment]
Is it possible to find more verbose error message?
What version of Qt do you have?

**Comment by username7778** [comment]
i know very little of mingw
i set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wextra") in the CMakeLists.txt witout a difference in the output.

my Qt version is 5.15.6  pacman -Sy mingw-w64-{i686,x86_64}-qt5
i will remove this version and install a newer one like in this wiki https://wiki.qt.io/Building_a_static_Qt_for_Windows_using_MinGW
